crypto/evp/p_lib.c

```diff
@@ -2176,7 +2176,15 @@ int EVP_PKEY_get_bn_param(const EVP_PKEY *pkey, const char *key_name,
         goto err;
     ret = OSSL_PARAM_get_BN(params, bn);
 err:
-    OPENSSL_free(buf);
+    if (buf) {
+        if (OSSL_PARAM_modified(params)) {
```

Comment:
for this inner if where the body of the if and else blocks are single lines, you can remove the {}